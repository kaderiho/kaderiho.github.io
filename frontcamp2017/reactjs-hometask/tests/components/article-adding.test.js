import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArticleAdding from '../../src/browser/components/article/article-adding';
import configureStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });

describe('<ArticleAdding />', () => {
    const mockStore = configureStore();
    let store, container;

    beforeEach(()=>{
        store = mockStore(initialState);
        container = shallow(<ArticleAdding store={store} /> )
    });

    it('Rendered a component', () => {
        expect(container.length).toEqual(1);
    });
});