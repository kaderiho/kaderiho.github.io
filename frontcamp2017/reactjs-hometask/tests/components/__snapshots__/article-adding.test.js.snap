// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleAdding UI testing/> Rendering component 1`] = `
<form
  className="addingTodoForm"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
    >
      Your article message
    </label>
    <input
      className="form-control"
      name="message"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
    >
      Author name
    </label>
    <input
      className="form-control"
      name="author"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-primary btn-lg"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
